apps/desktop-wallet/src/features/walletConnect/walletConnectContext.tsx

```diff
@@ -739,9 +519,13 @@ export const WalletConnectContextProvider: FC = ({ children }) => {
         sendUserRejectedResponse,
         sendSuccessResponse,
         sendFailureResponse,
-        refreshActiveSessions
+        refreshActiveSessions,
+        respondToWalletConnectWithError
       }}
     >
+      {sessionRequestEvent && isWalletUnlocked && (
+        <WalletConnectSessionRequestEventHandler sessionRequestEvent={sessionRequestEvent} />
```

Comment:
I understand why you created a component instead of a hook for this (to avoid running anything if it's not needed), but this highlights when of the limitation of React that is a bit frustrating to me: the rules of hooks. Anyway, no need to comment, just a little rant from me 😆

Reply:
Yes, exactly 💯